From 5a2698f605c5d7ff21ea3bac2fa757a6f4f42887 Mon Sep 17 00:00:00 2001
From: Rikardo-hm <rikard.olander@hostmobility.com>
Date: Thu, 30 Jan 2025 14:10:57 +0000
Subject: [PATCH 1/1] Device tree for V8 kernel 6.6
v1. Add our customization based on k3-am62-verdin-dev.
We have removed or disabled interfaces that we will not use( like sound, gpio-keys and eth2)
and add i2c handling, USB otg, digital in/out and som minor configuration changes.
Also add gpio names that we will use. There is still a lot of
TODOs in this file but this should be a good start.

v2. Add serialnumber, part-number and product-number.

V3. Add activate remoteproc for m4 processor.

V4. Fix LEDs and rest of the IOs is implemented and modem driver and gpio overlay added.
    Add support for gpio out and digital in.

v5. Enable support for analog in 2 enable.
    Fix usb-c cc control, bug wrong id implemented and add missing irq pin.
    Change UART1_CTSn to be controlled by userspace(could be driver or kernel controlled). Change so this pin is a output instead of a input.
    disable use of external temperature sensor, this is so far not mounted on any variant.

v6. Enable CTRL_WAKE1_MICO for wake on CAN.
    Fix default behaviour for CAN standby signals (gave 2mA lower power draw in suspend at 24VDC).
    Fix RS-485 RXEN signal so that it's held low until suspend.
    Add name for USB_OC in gpioinfo list.

v6.1. Change heartbeat LED to LED 1 to indicate stable release.

v7. Add accelerometer node, introduce i2c bus for mcu

V8. Adapt this patch to kernel 6.6
---
 arch/arm64/boot/dts/ti/Makefile               |   1 +
 .../arm64/boot/dts/ti/k3-am62-verdin-hmm.dtsi | 635 ++++++++++++++++++
 .../boot/dts/ti/k3-am625-verdin-wifi-hmm.dts  |  73 ++
 3 files changed, 709 insertions(+)
 create mode 100644 arch/arm64/boot/dts/ti/k3-am62-verdin-hmm.dtsi
 create mode 100644 arch/arm64/boot/dts/ti/k3-am625-verdin-wifi-hmm.dts

diff --git a/arch/arm64/boot/dts/ti/Makefile b/arch/arm64/boot/dts/ti/Makefile
index 10cba46afd19..43a00f6548c5 100644
--- a/arch/arm64/boot/dts/ti/Makefile
+++ b/arch/arm64/boot/dts/ti/Makefile
@@ -27,6 +27,7 @@ dtb-$(CONFIG_ARCH_K3) += k3-am625-verdin-nonwifi-mallow.dtb
 dtb-$(CONFIG_ARCH_K3) += k3-am625-verdin-nonwifi-yavia.dtb
 dtb-$(CONFIG_ARCH_K3) += k3-am625-verdin-wifi-dahlia.dtb
 dtb-$(CONFIG_ARCH_K3) += k3-am625-verdin-wifi-dev.dtb
+dtb-$(CONFIG_ARCH_K3) += k3-am625-verdin-wifi-hmm.dtb
 dtb-$(CONFIG_ARCH_K3) += k3-am625-verdin-wifi-ivy.dtb
 dtb-$(CONFIG_ARCH_K3) += k3-am625-verdin-wifi-mallow.dtb
 dtb-$(CONFIG_ARCH_K3) += k3-am625-verdin-wifi-yavia.dtb
diff --git a/arch/arm64/boot/dts/ti/k3-am62-verdin-hmm.dtsi b/arch/arm64/boot/dts/ti/k3-am62-verdin-hmm.dtsi
new file mode 100644
index 000000000000..9a467637b8af
--- /dev/null
+++ b/arch/arm64/boot/dts/ti/k3-am62-verdin-hmm.dtsi
@@ -0,0 +1,635 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/*
+ * Copyright 2023 Toradex
+ * Copyright 2024 SETEK Systems AB
+ *
+ * Common dtsi for Verdin AM62 SoM on Host monitor mini
+ */
+
+#include <dt-bindings/leds/common.h>
+
+/ {
+	/*	The leds are read from the highest to lowest position on the module when the PCB is in horizontal position.
+		Default: Heart beat on the lowest green LED (1) and wifi phy0rx on green LED (4). Can be turned off by system user or altered in a device tree patch.
+	*/
+	leds {
+		compatible = "gpio-leds";
+
+		green_led_1 {
+			color = <LED_COLOR_ID_GREEN>;
+			function = "led_1";
+			gpios = <&max7321 2 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "heartbeat";
+			default-state = "off";
+		};
+
+		red_led_1 {
+			color = <LED_COLOR_ID_RED>;
+			function = "led_1";
+			gpios = <&max7321 3 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+
+		green_led_2 {
+			color = <LED_COLOR_ID_GREEN>;
+			function = "led_2";
+			gpios = <&max7321 0 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+
+		red_led_2 {
+			color = <LED_COLOR_ID_RED>;
+			function = "led_2";
+			gpios = <&max7321 1 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+
+		green_led_3 {
+			color = <LED_COLOR_ID_GREEN>;
+			function = "led_3";
+			gpios = <&max7321 6 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+
+		red_led_3 {
+			color = <LED_COLOR_ID_RED>;
+			function = "led_3";
+			gpios = <&max7321 7 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+
+		green_led_4 {
+			color = <LED_COLOR_ID_GREEN>;
+			function = "led_4";
+			gpios = <&max7321 4 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0rx";
+			default-state = "off";
+		};
+
+		red_led_4 {
+			color = <LED_COLOR_ID_RED>;
+			function = "led_4";
+			gpios = <&max7321 5 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+
+
+	};
+	host_watchdog {
+		compatible = "host_watchdog";
+		status = "okay";
+		gpios = <&mcu_gpio0 1 GPIO_ACTIVE_HIGH>; //FORCE_RESET, not used by driver maybe in the future.
+		nwdog = <489>; //FORCE_RESET, used to select correct reset pin (see /sys/kernel/debug/gpio)
+
+	};
+	gpio-overlay {
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_dio_1>, <&pinctrl_dio_2a>;
+		compatible = "hostmobility,gpio_overlay_hmm";
+		status = "okay";
+		DIG-OUT-HI-gpios =
+					<&main_gpio1  1 GPIO_ACTIVE_HIGH>, /* UIO1_DOUT_SRC */
+					<&main_gpio0  88 GPIO_ACTIVE_HIGH>;
+
+		DIG-OUT-LO-gpios =
+				 	<&main_gpio1  2 GPIO_ACTIVE_HIGH>, /* UIO1_DOUT_SINK */
+					<&main_gpio0  87 GPIO_ACTIVE_HIGH>;
+		DIG-IN-gpios =
+					<&main_gpio1  3 GPIO_ACTIVE_HIGH>,
+		 			<&main_gpio1  0 GPIO_ACTIVE_HIGH>;
+
+		UIO_ANA_IN_20MA_EN-gpios = <&main_gpio1 4 GPIO_ACTIVE_HIGH>, <&main_gpio0 91 GPIO_ACTIVE_HIGH>; /* UIO1_ANA_IN_20MA_EN, UIO2_ANA_IN_20MA_EN (enable analog in 0-20mA measurement) */
+	};
+
+	gpio-quectel-boot {
+		pinctrl-names = "default";
+		pinctrl-0 =  <&pinctrl_modem>;
+		compatible = "hostmobility,modemcontroller";
+		status = "okay";
+		MDM_PWR_KEY-gpios =  <&main_gpio1  11 GPIO_ACTIVE_HIGH>;/*MDM_START*/
+		MDM_RESET-gpios =  <&main_gpio1  12 GPIO_ACTIVE_HIGH>;/*MDM_RST_MODEM_OVERRIDE*/
+		MDM_USB_BOOT-gpios =  <&main_gpio1  10 GPIO_ACTIVE_LOW>;/*nMDM_MODE_RECOV*/
+
+		//MDM_WAKEUP_IN-gpios =  <>;
+		//MDM_W_DISABLE-gpios =  <>;
+		//MDM_AP_READY-gpios =  <>;
+		//MDM_ANT_GNSS_PWR_EN-gpios =  <>;
+		//MDM_ANT_GNSS_SHORT-gpios =  <>;
+		MDM_STATUS-gpios =  <&main_gpio1  9 GPIO_ACTIVE_HIGH>;/*nMDM_OPERATIONAL ,STATUS*/
+		MDM_NET_STATUS-gpios =  <&main_gpio0  35 GPIO_ACTIVE_HIGH>; /*MDM_NET_ACTIVITY, NET_STATUS*/
+		//MDM_RINGO-gpios =  <>;
+		MDM_NET_MODE-gpios =  <&main_gpio0  37 GPIO_ACTIVE_HIGH>; /*MDM_LTE_REGISTRED, NET_MODE*/
+		//vdd-supply = <&reg_modem_4v>;
+		gpio-line-names = "MODEM_ENABLE_ON","MODEM_RESET","MODEM_POWER_ENABLE","MODEM_STATUS_ON", "MODEM_RECOVERY_ENABLE";
+	};
+
+	/* Enable Verdin CTRL_WAKE1_MICO# (SODIMM 252) */
+	/* wakeup-source for CAN 0, CAN 1 and RS-485 (RX) */
+	verdin_gpio_keys: gpio-keys {
+		status = "okay";
+	};
+};
+
+/* Verdin ETHs */
+&cpsw3g {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_rgmii1>;
+	status = "okay";
+};
+
+/* MDIO, shared by Verdin ETH_1 (On-module PHY) and removed not used Verdin ETH_2_RGMII */
+&cpsw3g_mdio {
+	status = "okay";
+};
+
+/* Verdin ETH_1 (On-module PHY) */
+&cpsw_port1 {
+	status = "okay";
+};
+
+
+/* Verdin I2C_1 */
+&main_i2c1 {
+	status = "okay";
+
+	max7321: gpio-expander@68 {
+		compatible = "maxim,max7321";
+		reg = <0x68>;
+		gpio-controller;
+		#gpio-cells = <2>;
+		gpio-line-names =
+				"green_led_2", "red_led_2",
+				"green_led_1", "red_led_1",
+				"green_led_4", "red_led_4",
+				"green_led_3", "red_led_3";
+		interrupt-controller;
+		#interrupt-cells = <2>;
+	};
+
+	/*USB-C controller*/
+	typec@1d {
+		compatible = "nxp,ptn5150";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_usb1_irq>;
+		interrupt-parent = <&main_gpio0>;
+		interrupts = <38 IRQ_TYPE_LEVEL_HIGH>;
+		reg = <0x1d>;
+		status ="okay";
+	};
+
+	eeprom@54 {
+		compatible = "st,24c02", "atmel,24c04";
+		reg = <0x54>;
+		status = "okay";
+		//vcc-supply = <&reg_base_3v3>;
+	};
+
+	/* HW confirmed, TMP103D has address 0x73 */
+	temperature-sensor@73 {
+		compatible = "ti,tmp103";
+		reg = <0x73>;
+		status = "disable";
+	};
+
+	/*
+	Add HSM module
+	hsm@48 {
+		compatible = "There is no driver";
+		reg = <0x48>;
+		status = "okay";
+	};
+	*/
+
+};
+
+&mcu_i2c0 {
+	status = "okay";
+	pinctrl-0 = <&pinctrl_mcu_i2c0>;
+
+	accelerometer@1e {
+		compatible = "st,lis2hh12";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_acc>;
+		reg = <0x1e>;
+		st,drdy-int-pin = <1>; /*Not working but set manually data ready 'i2cset -f -y 3 0x1e 0x22 1'*/
+		interrupt-parent = <&mcu_gpio0>;
+		interrupts = <2 IRQ_TYPE_EDGE_RISING>;
+		wakeup-source;
+		interrupt-names = "INT1";
+		status = "okay";
+	};
+};
+/* Verdin CAN_1 */
+&main_mcan0 {
+	status = "okay";
+};
+
+/* Verdin CAN_2 */
+&mcu_mcan0 {
+	status = "okay";
+};
+
+/* Verdin SPI_1 , not used right now but maybe in the future HW*/
+&main_spi1 {
+	status = "disabled";
+};
+
+/* Verdin UART_3 (debug console)*/
+&main_uart0 {
+	status = "okay";
+};
+
+/* Verdin UART_1, (RS485) */
+&main_uart1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart1_rs485>;
+	linux,rs485-enabled-at-boot-time;
+	rs485-rx-during-tx;
+	rs485-rts-active-low;
+	status = "okay";
+};
+
+/* Verdin UART_4 (m4 or not connected)*/
+&mcu_uart0 {
+	status = "disabled";
+};
+
+/* Verdin UART_2 (RS232)*/
+&wkup_uart0 {
+	status = "okay";
+};
+
+/* Verdin SD_1 , SD card TODO maybe change driver-strength-ohm*/
+&sdhci1 {
+	ti,driver-strength-ohm = <33>;
+	status = "okay";
+};
+
+/* Verdin USB_1 */
+&usbss0 {
+	status = "okay";
+};
+
+&usb0 {
+	status = "okay";
+	dr_mode = "otg";
+};
+
+/* Verdin USB_2 */
+&usbss1 {
+	status = "okay";
+};
+
+/*USB for modem*/
+&usb1 {
+	status = "okay";
+	dr_mode = "host";
+};
+
+/* Verdin PCIE_1_RESET# SODIM 244*/
+&verdin_pcie_1_reset_hog {
+	status = "okay";
+};
+
+&main_gpio0 {
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&pinctrl_ctrl_sleep_moci>, <&pinctrl_can_standby_normal>;
+	pinctrl-1 = <&pinctrl_ctrl_sleep_moci>, <&pinctrl_can_standby_sleep>;
+
+	gpio-line-names =
+		"SODIMM_52", /* 0 */
+		"",
+		"",
+		"SODIMM_56",
+		"PWR_GOOD",
+		"SODIMM_60",
+		"SODIMM_62",
+		"",
+		"",
+		"",
+		"", /* 10 */
+		"SODIMM_54",
+		"SODIMM_64",
+		"",
+		"",
+		"SODIMM_174",
+		"SODIMM_172",
+		"",
+		"",
+		"",
+		"", /* 20 */
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"SD_1_PWR_EN",
+		"SODIMM_21", /* 30 */
+		"SODIMM_256",
+		"WAKE_UP",
+		"",
+		"SODIMM_46",
+		"MDM_NET_ACTIVITY",
+		"SODIMM_218",
+		"MDM_LTE_REGISTRED",
+		"SODIMM_189",
+		"",
+		"SODIMM_216", /* 40 */
+		"SODIMM_220",
+		"SODIMM_222",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"", /* 50 */
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"", /* 60 */
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"", /* 70 */
+		"USB1_OC",
+		"USB2_OC",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"", /* 80 */
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"UIO2_DOUT_SINK",
+		"UIO2_DOUT_SRC",
+		"CAN2_STBY",
+		"CAN1_STBY", /* 90 */
+		"UIO2_ANA_IN_20MA_EN";
+
+	verdin_ctrl_sleep_moci: ctrl-sleep-moci-hog {
+		gpio-hog;
+		/* Verdin CTRL_SLEEP_MOCI# (SODIMM 256) */
+		gpios = <31 GPIO_ACTIVE_HIGH>;
+		line-name = "CTRL_SLEEP_MOCI#";
+		output-high;
+	};
+	can0_standby_normal: can0-standby-normal-hog {
+		gpio-hog;
+		gpios = <90 GPIO_ACTIVE_HIGH>;
+		line-name = "CAN0_STBY";
+		output-low;
+		status = "okay";
+	};
+	can1_standby_normal: can1-standby-normal-hog {
+		gpio-hog;
+		gpios = <89 GPIO_ACTIVE_HIGH>;
+		line-name = "CAN1_STBY";
+		output-low;
+		status = "okay";
+	};
+};
+
+
+&main_gpio1 {
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart1_cts>;
+	gpio-line-names =
+		"UIO2_DIG_IN", /* 0 */
+		"UIO1_DOUT_SRC",
+		"UIO1_DOUT_SINK",
+		"UIO1_DIG_IN",
+		"UIO1_ANA_IN_20MA_EN",
+		"",
+		"",
+		"RS485_RX_EN",
+		"",
+		"nMDM_OPERATIONAL",
+		"nMDM_MODE_RECOV", /* 10 */
+		"MDM_START",
+		"MDM_RST_MODEM_OVERRIDE",
+		"",
+		"",
+		"SODIMM_15",
+		"SODIMM_16",
+		"SODIMM_19",
+		"SODIMM_66",
+		"SODIMM_161",
+		"", /* 20 */
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"", /* 30 */
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"", /* 40 */
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"SODIMM_17",
+		"SODIMM_155", /* 50 */
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"", /* 60 */
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"", /* 70 */
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"", /* 80 */
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"";
+	rs485_temp_rxen_hog: rs485-temp-rxen-hog {
+		gpio-hog;
+		/* RS485_RX_EN is always low in Linux, except in standby where it's held high using an external pull up. */
+		gpios = <7 GPIO_ACTIVE_LOW>;
+		line-name = "RS485_RX_EN";
+		output-high;
+		status = "okay";
+	};
+};
+
+&mcu_gpio0 {
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pcie_1_reset>;
+	gpio-line-names =
+		"SODIMM_244_PCIE_RESET",
+		"FORCE_RESET",
+		"SODIMM_208",
+		"ACCM_RDY_IRQ",
+		"SODIMM_212",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"",
+		"";
+
+	verdin_pcie_1_reset_hog: pcie-1-reset-hog {
+		gpio-hog;
+		/* Verdin PCIE_1_RESET# (SODIMM 244) */
+		gpios = <0 GPIO_ACTIVE_LOW>;
+		line-name = "PCIE_1_RESET#";
+		output-low;
+		status = "disabled";
+	};
+
+};
+
+&main_pmx0 {
+	/*Note!: Select am62xSIP in ti sysconfig to get more information about the pin modes.*/
+	pinctrl_dio_1: pinctrl-dio-1-default-pins {
+		pinctrl-single,pins = <
+			AM62X_IOPAD(0x178, PIN_INPUT, 7) /* (AC20) RGMII2_TD3, GPIO1_0    */ /* SODIMM 215 */ /*UIO2_DIG_IN*/
+			AM62X_IOPAD(0x017c, PIN_OUTPUT, 7) /* (W19) RGMII2_RX_CTL.GPIO1_1 */ /* UIO1_DOUT_SRC */
+			AM62X_IOPAD(0x0180, PIN_OUTPUT, 7) /* (V18) RGMII2_RXC.GPIO1_2 */ /* UIO1_DOUT_SINK */
+			AM62X_IOPAD(0x0184, PIN_INPUT, 7) /* (W18) RGMII2_RD0.GPIO1_3 */ /* UIO1_DIG_IN */
+			AM62X_IOPAD(0x0188, PIN_OUTPUT, 7) /* (Y20) RGMII2_RD1.GPIO1_4 */ /* UIO1_ANA_IN_20MA_EN */
+		>;
+	};
+	pinctrl_dio_2a: pinctrl-dio-2a-default-pins {
+		pinctrl-single,pins = <
+			AM62X_IOPAD(0x0164, PIN_OUTPUT, 7) /* (Y21) RGMII2_TX_CTL.GPIO0_87 */ /* UIO2_DOUT_SINK */
+			AM62X_IOPAD(0x0168, PIN_INPUT, 7) /* (Y18) RGMII2_TXC.GPIO0_88 */ /* UIO2_DOUT_SRC */
+			AM62X_IOPAD(0x174, PIN_OUTPUT, 7) /* (AD21) RGMII2_TD2.GPIO0_91    */ /* SODIMM 217 */ /*UIO2_ANA_IN_20MA_EN*/
+		>;
+	};
+
+	pinctrl_can_standby_normal: pinctrl-can-standby-default-pins {
+		pinctrl-single,pins = <
+			AM62X_IOPAD(0x16c, PIN_OUTPUT, 7) /*  (Y18) RGMII2_TD0.GPIO0_89 */ /* SODIMM 221 */ /*CAN2_STBY (active normal mode on low)*/
+			AM62X_IOPAD(0x170, PIN_OUTPUT, 7) /* (AA18) RGMII2_TD1.GPIO0_90 */ /* SODIMM 219 */ /*CAN1_STBY */
+		>;
+	};
+	pinctrl_can_standby_sleep: pinctrl-can-standby-sleep-pins {
+		pinctrl-single,pins = <
+			AM62X_IOPAD(0x16c, PIN_INPUT_PULLUP, 7) /*  (Y18) RGMII2_TD0.GPIO0_89 */ /* SODIMM 221 */ /*CAN2_STBY (active standby on high or floating)*/
+			AM62X_IOPAD(0x170, PIN_INPUT_PULLUP, 7) /* (AA18) RGMII2_TD1.GPIO0_90 */ /* SODIMM 219 */ /*CAN1_STBY */
+		>;
+	};
+
+	pinctrl_modem: pinctrl-modem-default-pins {
+		pinctrl-single,pins = <
+			AM62X_IOPAD(0x01a4, PIN_OUTPUT,  7) /* (B20) MCASP0_ACLKX.GPIO1_11 */ /* SODIMM 30 */ /*MDM_START*/
+			AM62X_IOPAD(0x01a8, PIN_OUTPUT,  7) /* (D20) MCASP0_AFSX.GPIO1_12  */ /* SODIMM 32 */ /*MDM_RST_MODEM_OVERRIDE*/
+			AM62X_IOPAD(0x01a0, PIN_OUTPUT, 7) /* (E18) MCASP0_AXR0  */ /* SODIMM 34 */ /*nMDM_MODE_RECOV*/
+			AM62X_IOPAD(0x019c, PIN_INPUT_PULLUP,  7) /* (B18) MCASP0_AXR1  */ /* SODIMM 36 */ /*nMDM_OPERATIONAL ,STATUS*/
+			AM62X_IOPAD(0x0090, PIN_INPUT_PULLUP, 7) /* (M24) GPMC0_BE0n_CLE.GPIO0_35 */ /* SODIMM 42 */ /*MDM_NET_ACTIVITY, NET_STATUS*/
+			AM62X_IOPAD(0x0098, PIN_INPUT,  7) /* (U23) GPMC0_WAIT0.MCASP1_AFSX     */ /* SODIMM 44 */ /*MDM_LTE_REGISTRED, NET_MODE , can not be pull-up before start!*/
+		>;
+	};
+	/* Verdin QSPI_1_IO1 as GPIO (conflict with Verdin QSPI_1 interface) */
+	pinctrl_power_good_gpio: main-power-good-pins-default {
+		pinctrl-single,pins = <
+			AM62X_IOPAD(0x0010, PIN_INPUT, 7) /* (G24) OSPI0_D1.GPIO0_4 */ /* SODIMM 58 */ /*PWR_GOOD, */
+			/*AM62X_MCU_IOPAD(0x0004, PIN_INPUT, 7) /* (B8) MCU_SPI0_CS1.MCU_GPIO0_1 */ /* SODIMM 206 */ /*SOM_FORCE_RESET used by M4*/
+		>;
+	};
+	/* Verdin UART_1 change to not use CTS here for now*/
+	pinctrl_uart1_rs485: main-uart1-rs485-pins-default {
+		pinctrl-single,pins = <
+			AM62X_IOPAD(0x0198, PIN_OUTPUT,       2) /* (A19) MCASP0_AXR2.UART1_RTSn */ /* SODIMM 133 */
+			AM62X_IOPAD(0x01ac, PIN_INPUT_PULLUP, 2) /* (E19) MCASP0_AFSR.UART1_RXD  */ /* SODIMM 129 */
+			AM62X_IOPAD(0x01b0, PIN_OUTPUT,       2) /* (A20) MCASP0_ACLKR.UART1_TXD */ /* SODIMM 131 */
+		>;
+	};
+	pinctrl_uart1_cts: main-uart1-cts-pins-default {
+		pinctrl-single,pins = <
+			AM62X_IOPAD(0x0194, PIN_OUTPUT, 7) /* (B19) MCASP0_AXR3.UART1_CTSn */ /* SODIMM 135 , GPIO1_7, hogged low until we fix a driver.*/
+		>;
+	};
+	pinctrl_usb1_irq: usb1-irq-pins-default {
+		pinctrl-single,pins = <
+			AM62X_IOPAD(0x009c, PIN_INPUT, 7) /* (V25) GPMC0_WAIT1.GPIO0_38 */ /* GPIO_SODIMM_189 */
+		>;
+	};
+	/* Accelerometer IRQ line */
+	pinctrl_acc: acc_grp {
+		pinctrl-single,pins = <
+			AM62X_MCU_IOPAD(0x000c, PIN_INPUT, 7) /* (D9) MCU_SPI0_D0.MCU_GPIO0_3 */ /* GPIO_SODIMM 210 */
+		>;
+	};
+
+};
diff --git a/arch/arm64/boot/dts/ti/k3-am625-verdin-wifi-hmm.dts b/arch/arm64/boot/dts/ti/k3-am625-verdin-wifi-hmm.dts
new file mode 100644
index 000000000000..65fbb517c1a8
--- /dev/null
+++ b/arch/arm64/boot/dts/ti/k3-am625-verdin-wifi-hmm.dts
@@ -0,0 +1,73 @@
+/*
+ * SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+ *
+ * Copyright 2023 Toradex
+ * Copyright 2024 SETEK Systems AB
+ *
+ */
+
+/dts-v1/;
+
+#include "k3-am625.dtsi"
+#include "k3-am62-verdin.dtsi"
+#include "k3-am62-verdin-wifi.dtsi"
+#include "k3-am62-verdin-hmm.dtsi"
+
+/ {
+	model = "Toradex Verdin AM62 WB on Host Monitor Mini board";
+	compatible = "SETEK Systems AB,verdin-am62-hmm",
+		     "toradex,verdin-am62-wifi",
+		     "toradex,verdin-am62",
+		     "ti,am625";
+
+	chosen {
+		stdout-path = "serial2:115200n8";
+		serial-number = "103101XXXXX";
+		part-number = "HMP1031-0001";
+		product-number = "hm20";
+
+	};
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		rtos_ipc_memory_region: ipc-memories@9c800000 {
+			compatible = "shared-dma-pool";
+			reg = <0x00 0x9c800000 0x00 0x00300000>;
+			no-map;
+		};
+
+		mcu_m4fss_dma_memory_region: m4f-dma-memory@9cb00000 {
+			compatible = "shared-dma-pool";
+			reg = <0x00 0x9cb00000 0x00 0x100000>;
+			no-map;
+		};
+
+		mcu_m4fss_memory_region: m4f-memory@9cc00000 {
+			compatible = "shared-dma-pool";
+			reg = <0x00 0x9cc00000 0x00 0xe00000>;
+			no-map;
+		};
+	};
+};
+
+&mailbox0_cluster0 {
+	status = "okay";
+
+	mbox_m4_0: mbox-m4-0 {
+		ti,mbox-rx = <0 0 0>;
+		ti,mbox-tx = <1 0 0>;
+	};
+};
+
+&mcu_m4fss {
+       mboxes = <&mailbox0_cluster0 &mbox_m4_0>;
+       memory-region = <&mcu_m4fss_dma_memory_region>,
+                       <&mcu_m4fss_memory_region>;
+};
+
+/* Verdin UART_4 (m4 or TBD not connected)*/
+&mcu_uart0 {
+        status = "reserved";
+};
\ No newline at end of file
-- 
2.25.1

